classpath/java/lang/invoke/LambdaMetafactory.java

```diff
@@ -290,20 +319,48 @@ public static CallSite metafactory(MethodHandles.Lookup caller,
                                      MethodType instantiatedMethodType)
     throws LambdaConversionException
   {
-    byte[] classData = makeLambda(invokedName, invokedType, methodType, methodImplementation);
-    
-    try {
-      return new CallSite
-        (new MethodHandle
-         (MethodHandle.REF_invokeStatic, invokedType.loader, Classes.toVMMethod
-          (avian.SystemClassLoader.getClass
-           (avian.Classes.defineVMClass
-            (invokedType.loader, classData, 0, classData.length))
-           .getMethod("make", invokedType.parameterArray()))));
-    } catch (NoSuchMethodException e) {
-      AssertionError error = new AssertionError();
-      error.initCause(e);
-      throw error;
+    byte[] classData = makeLambda(invokedName, invokedType, methodType, methodImplementation, emptyInterfaceList);
+    return makeCallSite(invokedType, classData);
+  }
+
+  public static CallSite altMetafactory(MethodHandles.Lookup caller,
+                                        String invokedName,
+                                        MethodType invokedType,
+                                        Object... args) throws LambdaConversionException {
+    // Behaves as if the prototype is like this:
+    //
+    // CallSite altMetafactory(
+    //    MethodHandles.Lookup caller,
+    //    String invokedName,
+    //    MethodType invokedType,
+    //    MethodType methodType,
+    //    MethodHandle methodImplementation,
+    //    MethodType instantiatedMethodType,
+    //    int flags,
+    //    int markerInterfaceCount,  // IF flags has MARKERS set
+    //    Class... markerInterfaces, // IF flags has MARKERS set
+    //    int bridgeCount,           // IF flags has BRIDGES set
+    //    MethodType... bridges      // IF flags has BRIDGES set
+    //  )
+
+    MethodType methodType = (MethodType) args[0];
+    MethodHandle methodImplementation = (MethodHandle) args[1];
+
+    int flags = (Integer) args[3];
+    boolean serializable = (flags & FLAG_SERIALIZABLE) != 0;
+    if ((flags & FLAG_MARKERS) != 0)
+      throw new UnsupportedOperationException("FLAG_MARKERS");
```

Comment:
Do you have an idea of what this is used for and what might be needed to implement it?  If so, it might be handy to leave a comment to that effect.